### PR TITLE
docs: add documentation

### LINKED ISSUE #209: Update docstrings + remove types
We should update docstrings (some may not be descriptive enough) and also remove types since they're defined in the signature

### PR DESCRIPTION
**Pull request by jsstevenson**
close #209 , address #11 

Because of how many different things CST can do, I tried to embed a lot of explanation/instructions directly in the API reference rather than writing it out separately. 

* The sphinx docs test action that I'd been using is unfortunately not happy about `psycopg2` and is also unhappy with every one of my attempts to install the binary manually. I've gone ahead and just done a basic manual test, but the logging won't be as pretty.
* What do we think we should be saying about VRS conformance? Maybe just say that this version is consistent with 1.x? We could also leave it light for now and return in #219 (could also wait until VRS 2 alpha branch is merged)
* Left a few small TODOs/question marks scattered where I was unsure if I was giving a correct explanation
* I think it might be cool to add an example script that uses some of the functions to do something cool (maybe some code extracted from the Variation Normalizer). Not urgently trying to get that added now unless anyone has a good idea.

### CODE REVIEW

**Comment by jsstevenson** [comment]
TODOs EOD 2023/12/05

* Merge from VRS 2.x version, address in documentation
* Describe layout, basics of mappers vs handlers etc
* Different default port?
* Add examples to class docstrings

**Comment by jsstevenson** [comment]
TODO EOD 20231208: look at MANETranscript class docstrings

File: cool_seq_tool/mappers/exon_genomic_coords.py [hunk 1]
```diff
@@ -63,7 +78,25 @@ async def transcript_to_genomic_coordinates(
         **kwargs,
     ) -> GenomicDataResponse:
         """Get genomic data given transcript data.
-        Will use GRCh38 coordinates if possible
+
+        By default, inputs are assumed to be in GRCh38 where they aren't otherwise
+        precisely defined.  # TODO is this correct?
```
Comment (korikuzma) [inline cool_seq_tool/mappers/exon_genomic_coords.py:83]: I can make sure to address this in #224

File: cool_seq_tool/mappers/mane_transcript.py [hunk 2]
```diff
@@ -574,23 +590,47 @@ async def get_longest_compatible_transcript(
...
+        ...     start_annotation_layer=AnnotationLayer.PROTEIN,
+        ...     residue_mode=ResidueMode.INTER_RESIDUE,
+        ...     mane_transcripts=mane_transcripts,
+        ... )
+        >>> result.refseq
+        'NP_001365396.1'
+
+        Try GRCh38 first, then GRCh37.
+
+        # TODO example for this?
```
Comment (korikuzma) [inline cool_seq_tool/mappers/mane_transcript.py:619]: Did you want to add examples in here or new issue?

File: cool_seq_tool/mappers/mane_transcript.py [hunk 3]
```diff
@@ -1026,22 +1076,52 @@ async def g_to_mane_c(
...
+        ...     None,
+        ...     gene="EGFR"
+        ... )
+        >>> type(result)
+        cool_seq_tool.mappers.mane_transcript.CdnaRepresentation
+        >>> result.status
+        <TranscriptPriority.MANE_SELECT: 'mane_select'>
+
+        If no matching MANE cDNA representation is found, instead returns genomic
+        coordinates on GRCh38.
```
Comment (korikuzma) [inline cool_seq_tool/mappers/mane_transcript.py:1103]: genomic data is returned when no input gene is provided

File: cool_seq_tool/sources/transcript_mappings.py [hunk 4]
```diff
@@ -7,7 +7,17 @@
...
+    """Provide mappings between gene symbols and RefSeq + Ensembl transcript accessions.
+
+    Uses ``LRG_RefSeqGene`` and ``transcript_mappings.csv``, which will automatically
+    be acquired if they aren't already available. See the
+    :ref:`configuration <configuration>` section in the documentation for information
+    about manual acquisition of data.
+
+    In general, this class's methods expect to receive NCBI gene symbols, so users
+    should be careful about the sourcing of their input in cases where terms are
+    conflicted or ambiguous.  # TODO is that true?
```
Comment (korikuzma) [inline cool_seq_tool/sources/transcript_mappings.py:19]: I think that's right. We should come back to this. I think we swap between HGNC and NCBI. We removed the gene normalizer. Maybe we should add it back for this

**Comment by jsstevenson** [comment]
Remaining TODO

- [ ] Address comments above
- [ ] Refine README
- [ ] Fill out docs front page
- [ ] Readthedocs deployment/integration
- [ ] Refine language about async/maybe better examples?

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    git clone https://github.com/genomicmedlab/cool-seq-tool
+    cd cool-seq-tool
+
+Then initialize the Pipenv environment: ::
+
+    pipenv update
+    pipenv install --dev
+    pipenv shell
+
+Alternatively, use a virtual environment and install :ref:`all dependency groups<dependency-groups>`: ::
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply docs/source/contributing.rst:26]: #231

File: cool_seq_tool/sources/uta_database.py [hunk 6]
```diff
@@ -696,11 +717,27 @@ async def get_mane_c_genomic_data(
...
+        ... )
+        >>> result["gene"]
+        'BRAF'
+        >>> result["alt_ac"]
+        'NC_000007.14'
+
+        :param ac: MANE transcript accession
+        :param alt_ac: NC accession. Used to triangulate on correct genomic data. Can
+            be set to ``None`` if unavailable.
+        :param start_pos: Genomic start position change  # TODO do we still want to say "change" here?
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply cool_seq_tool/sources/uta_database.py]: Yeah I assumed this was leftover from when it was more tightly coupled with the normalizer

File: cool_seq_tool/sources/transcript_mappings.py (hunk 4 above)
Comment (jsstevenson) [reply cool_seq_tool/sources/transcript_mappings.py]: #232

File: cool_seq_tool/mappers/mane_transcript.py [hunk 7]
```diff
@@ -498,10 +511,13 @@ def _validate_index(
     ) -> bool:
         """Validate that positions actually exist on accession
 
-        :param str ac: Accession
-        :param Tuple[int, int] pos: Start position change, End position change
-        :param int coding_start_site: coding start site for accession
-        :return: `True` if positions exist on accession. `False` otherwise
+        # TODO example
+        even though it's not public (it should be?)
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply cool_seq_tool/mappers/mane_transcript.py]: It's used here: https://github.com/cancervariants/variation-normalization/blob/da230a5d3e0f44df1971589a42ff0a0dcb2e4937/variation/gnomad_vcf_to_protein_variation.py#L236

File: docs/source/transcript_selection.rst [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+.. _transcript_compatibility:
+
+Transcript compatibility
+------------------------
+
+The following validation checks are performed to determine compatibility of a transcript position:
+
+* The position exists on an accession
+* "Validating reference sequences"  # TODO what does this mean?
+* "Validating exon structure"  # TODO what does this mean?
```
Comment (jsstevenson) [inline docs/source/transcript_selection.rst:19]: @korikuzma can you clarify these two lines? They're taken from the previous markdown docs file

File: cool_seq_tool/mappers/mane_transcript.py [hunk 9]
```diff
@@ -574,23 +588,48 @@ async def get_longest_compatible_transcript(
...
+        ...     residue_mode=ResidueMode.INTER_RESIDUE,
+        ...     mane_transcripts=mane_transcripts,
+        ... )
+        >>> result.refseq
+        'NP_001365396.1'
+
+        If unable to find a match on GRCh38, this method will then attempt to drop down
+        to GRCh37.
+
+        # TODO example for inputs that demonstrate this?
```
Comment (jsstevenson) [inline cool_seq_tool/mappers/mane_transcript.py:618]: @korikuzma do you have an example set of inputs handy for this?

File: docs/source/transcript_selection.rst (hunk 8 above)
Comment (korikuzma) [reply docs/source/transcript_selection.rst:19]: IIRC validating reference sequences means ensuring actual reference sequence matches expected reference sequence for a given ac/pos. Exon structure has to deal with ensuring exon numbers are the same.

File: cool_seq_tool/mappers/mane_transcript.py (hunk 9 above)
Comment (korikuzma) [reply cool_seq_tool/mappers/mane_transcript.py:618]: Not off the top of my head. We can make a separate issue if needed.

Comment (jsstevenson) [reply cool_seq_tool/mappers/mane_transcript.py:618]: (sorry, this duplicates a comment you had above. Probably not necessary if there isn't a good example available -- I was just trying to illustrate different code branches based on input)

File: docs/source/transcript_selection.rst (hunk 8 above)
Comment (jsstevenson) [reply docs/source/transcript_selection.rst]: 🙌

File: cool_seq_tool/mappers/mane_transcript.py (hunk 9 above)
Comment (korikuzma) [reply cool_seq_tool/mappers/mane_transcript.py:618]: No worries. It is good to have an example and see if our test suite covers it. I just don't know off the top of my head and would have to dig into the test data

**Review by korikuzma — APPROVED** [review]

**Pull request merged**